### PR TITLE
Auto Orientation issue fixed

### LINKED ISSUE #18338: Auto orientation is not applied on responsive friendly image
### Required System information

- Node.js version: 10.1.0
- NPM version: 18.17.0
- Strapi version: 4.12.5
- Database: sqlite
- Operating system: win32-x64 (window10)
- Is your project Javascript or Typescript: Javascript

### Describe the bug

Auto orientation is not applied on responsive friendly image(thumbnail, small, medium, large size)

Media Library Settings: 
- Responsive friendly upload: TRUE
- Size optimization: FALSE
- Auto orientation: TRUE

Upload provider : aws-s3 (But, I think it is not related to this issue)

### Steps to reproduce the behavior

1. Click on Add new assets on 'Media Library' page
2. Upload image that have EXIF orientation property
3. Result : Auto orientation is only applied on original image not other image(thumbnail, small, medium, large size)

### Expected behavior

Auto orientation should be applied on all of images including responsive friendly image.

### Screenshots

**original image**

**image uploaded on s3**

**responsive friendly image(large) uploaded on s3**

### Code snippets

none

### Additional context

It seems to generating responsive images before 'auto orientation' is applied on image

generating responsive image: 
https://github.com/strapi/strapi/blob/0608d433f0396c305dadbe7300c1c9715869b34b/packages/core/upload/server/services/upload.js#L210

auto orientation: 
https://github.com/strapi/strapi/blob/0608d433f0396c305dadbe7300c1c9715869b34b/packages/core/upload/server/services/image-manipulation.js#L87

Thank you for your support!

### PR DESCRIPTION
**Pull request by PrathameshGujar08**
### What does it do?

The issue as being said **Auto orientation is not applied on responsive friendly image** was that If one uploads a image the orientation was changed or thumbnail view of the image did not align with the actual image uploaded.
As I found the resize function of sharp() results in loss of data which was responsible for such behaviour thus I retained the EXIF data (you can see it in the changes made in the code).

### Why is it needed?

It is needed to store the data as it was meant to be without any changes being made to it.

### How to test it?

The setup of the environment is same and to test it the original image which was being used to test can be used again. From my side I used some other images clicked from my phone with different camera orientation and some images were dowloaded from the internet.

### Related issue(s)/PR(s)

Fixes #18338

### CODE REVIEW

**Comment by Boegie19** [comment]
fixes https://github.com/strapi/strapi/issues/18338

File: packages/core/upload/server/services/image-manipulation.js [hunk 1]
```diff
@@ -97,10 +96,10 @@ const optimize = async (file) => {
     const transformer = sharp();
     // reduce image quality
     transformer[format]({ quality: sizeOptimization ? 80 : 100 });
-    // rotate image based on EXIF data
-    if (autoOrientation) {
```
Comment (Boegie19) [inline packages/core/upload/server/services/image-manipulation.js:101]: removing autoOrientation flag is not the solution.
Goal was to figure out why autoOrientation is set for A but not for B.
aka it should rotate everything or noting but this flag needs to stay intact.

**Review by Boegie19 — CHANGES_REQUESTED** [review]

**Review by DKdev117 — COMMENTED** [review]
Not right you have changed code and maybe  it is not the right solution

**Comment by PrathameshGujar08** [comment]
> @PrathameshGujar08 Not right you have changed code and maybe it is not the right solution
Did you get any lead over it bro? I would like to discuss it over Discord if you are fine with it

**Comment by PrathameshGujar08** [comment]
@Boegie19  Can you review it again? The exif data seems to be lost during sharp.resize() as I read online. So I retained the metadata while resizing which now shows the correct behaviour.

**Comment by Boegie19** [comment]
@PrathameshGujar08 It works and nice spot. mind making the PR ready to review?

**Review by Boegie19 — APPROVED** [review]
Tested it and solves the issue.

**Comment by ch-jwoo** [comment]
@Boegie19 
Thank you for your hard work

When are you going to merge this pr?
Is there any remain process?

**Comment by PrathameshGujar08** [comment]
> Tested it and solves the issue.

Any update for this?

**Comment by alexandrebodin** [comment]
> [!IMPORTANT] 
> With the release of Strapi 5, we have lifted the code freeze. To ensure we can effectively review and assist with contributions, we are now only accepting PRs targeting version 5 (the develop branch).<br/> 
> As your PR targets version 4, it has been automatically closed. However, we would greatly appreciate it if you could update your contribution to align with version 5 and resubmit!

**Pull request closed without merging**